sdk/typescript/src/types/common.ts

```diff
@@ -26,17 +26,23 @@ export type ObjectOwner =
 // source of truth is
 // https://github.com/MystenLabs/sui/blob/acb2b97ae21f47600e05b0d28127d88d0725561d/crates/sui-types/src/base_types.rs#L171
 const TX_DIGEST_LENGTH = 32;
-// taken from https://rgxdb.com/r/1NUN74O6
-const VALID_BASE64_REGEX =
-  /^(?:[a-zA-Z0-9+\/]{4})*(?:|(?:[a-zA-Z0-9+\/]{3}=)|(?:[a-zA-Z0-9+\/]{2}==)|(?:[a-zA-Z0-9+\/]{1}===))$/;
 
+/** Base58 string representing the transaction digest */
 export function isValidTransactionDigest(
   value: string
 ): value is TransactionDigest {
-  return (
-    new Base64DataBuffer(value).getLength() === TX_DIGEST_LENGTH &&
-    VALID_BASE64_REGEX.test(value)
-  );
+  let buffer;
+  try {
+    buffer = new Base58DataBuffer(value);
+  } catch (e) {
+    try {
+      // For backwards compatibility, we also accept Base64 strings
```

Comment:
I would pass in a `serializationFmt: 'base64' | 'base58'` to this method. The application will choose which one to use depending on API version

Reply:
revised